compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/mm_npu4.cc

```diff
@@ -158,14 +159,119 @@ void matmul_vectorized_8x8x8_bf16_bf16_f32(const bfloat16 *__restrict pA,
+          aie::accum<acc32, size_C>  acc_C10 = load_v64<int32, acc32>(pC1);
+          aie::accum<acc32, size_C>  acc_C11 = load_v64<int32, acc32>(pC1 + size_C * rowA);
+
+          MMUL C00(acc_C00);
+          MMUL C01(acc_C01);
+          MMUL C10(acc_C10);
+          MMUL C11(acc_C11);
+
+          C00.mac(A0, B0);
+          C01.mac(A0, B1);
+          C10.mac(A1, B0);
+          C11.mac(A1, B1);
+
+          for (unsigned i = 1; i < colA; ++i)
+            chess_prepare_for_pipelining chess_loop_range(7, ) {
+              A0 = aie::load_v<size_A>(pA0);
+              pA0 += rowA * size_A;
+              A1 = aie::load_v<size_A>(pA1);
+              pA1 += rowA * size_A;
+              B0 = aie::load_v<size_A>(pB0);
+              pB0 += size_B;
+              B1 = aie::load_v<size_A>(pB1);
+              pB1 += size_B;
+
+              C00.mac(A0, B0);
+              C01.mac(A0, B1);
+              C10.mac(A1, B0);
+              C11.mac(A1, B1);
+          }
+
+          v32int32 * __restrict pOut00 = (v32int32 *) pC0;
+          *pOut00++ = (v32int32)(C00.to_accum().template extract<32>(0));
+          *pOut00++ = (v32int32)(C00.to_accum().template extract<32>(1));
+          pC0 += size_C * rowA;
+
+          v32int32 * __restrict pOut01 = (v32int32 *) pC0;
+          *pOut01++ = (v32int32)(C01.to_accum().template extract<32>(0));
+          *pOut01++ = (v32int32)(C01.to_accum().template extract<32>(1));
+          pC0 += size_C * rowA;
+
+          v32int32 * __restrict pOut10 = (v32int32 *) pC1;
+          *pOut10++ = (v32int32)(C10.to_accum().template extract<32>(0));
+          *pOut10++ = (v32int32)(C10.to_accum().template extract<32>(1));
+          pC1 += size_C * rowA;
+
+          v32int32 * __restrict pOut11 = (v32int32 *) pC1;
+          *pOut11++ = (v32int32)(C11.to_accum().template extract<32>(0));
+          *pOut11++ = (v32int32)(C11.to_accum().template extract<32>(1));
+          pC1 += size_C * rowA;
+        }
+    }
+}
+
+template <unsigned m, unsigned k, unsigned n>
+void matmul_vectorized_8x8x8_i8_i8_i32(const int8 *__restrict pA,
+                                      unsigned offsetA,
+                                      const int8 *__restrict pB,
+                                      unsigned offsetB, int32 *__restrict pC,
+                                      unsigned offsetC) {
+  constexpr int r = 8;
+  constexpr int s = 8;
+  constexpr int t = 8;
+  static_assert(m / r > 0);
+  static_assert(k / s > 0);
+  static_assert(n / t > 0);
+  return matmul_vectorized_i8_i32<m / r, k / s, n / t, r, s, t>
+      (pA, offsetA, pB, offsetB, pC, offsetC);
+}
+
 extern "C" {
 
 #define matmul_combos(X, M, N, K)                                     \
   X(bfloat16, bf16, bfloat16, bf16, float, f32, M, N, K, 8, 8, 8)
 
+#define matmul_combos_i8(X, M, N, K)                                  \
+  X(int8, i8, int8, i8, int32, i32, M, N, K, 8, 8, 8)
```

Comment:
You can just add this to the previous `matmul_combo`.

Reply:
No, because then it tries to generate this variation: `matmul_combos(matmul_vectorized_c_func, 16, 8, 32)`, but because `N == 8` and the intrinsic does `8x8x8` as well, this throws an error. I think the minimum in every dimension is 16.